### DISCUSSION TITLE
Integrating PSAI with powershell script to evaluate data in a text file

### DISCUSSION THREAD

**Discussion by jmalco-bp** [dfinke/PSAI#28] [General]
I currently have a script to generate a list of update IDs that were attempted and successfully installed from WindowsUpdate.log. I can bring the list into Copilot and ask it to give me the Microsoft KBs and the description but I want to integrate this into the powershell script. Can PSAI do this if I dump the update IDs to a file or if I put the PSAI query in a function to run on each object in an array?

**Comment by dfinke** [comment]
@jmalco-bp Great question! 

With PSAI, you’ve got a powerful tool that keeps you in your flow, but when it comes to real-time data access, PSAI Agent takes it to another level. This premium module, currently exclusive, allows you to fetch and interact with up-to-the-minute information directly in PowerShell.

I’m hosting premium livestreams where I’ll be diving deep into what PSAI Agent can do. RSVP, and you’ll not only get the recording but also early access to the module itself.

Interested? https://www.meetup.com/nycpowershellmeetup/events/302816941

Here’s a quick example of what it can do:

```powershell
$list = @"
 KB5039747
 KB5038325
 KB5037331
"@

$agent = New-Agent -Tools (New-TavilyAITool)

$agent | Get-AgentResponse @"
lookup and provide details for each KB - in a table

$($list)
"@
```


Here are the details for the requested KB updates presented in a table format:

| KB Number    | Title                                           | Description                                                                                      | Published Date | Size     | Download Link                                                                                              |
|--------------|-------------------------------------------------|--------------------------------------------------------------------------------------------------|----------------|----------|------------------------------------------------------------------------------------------------------------|
| KB5039747    | Cumulative Update Package 28 for SQL Server 2019 | This cumulative update addresses issues and includes improvements for SQL Server 2019.          | 08/01/2024     | 880.2 MB | [Download KB5039747](https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=adaf355c-e3cf-4286-8230-af4de5ce739d)        |
| KB5038325    | Cumulative Update Package 14 for SQL Server 2022 | This cumulative update includes various fixes and improvements for SQL Server 2022.            | 07/23/2024     | 442.4 MB | [Download KB5038325](https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=14ce5b30-ce8d-4058-83d0-285932673e23)        |
| KB5037331    | Cumulative Update Package 27 for SQL Server 2019 | This update includes new features, fixes, and improvements for SQL Server 2019.                | 06/13/2024     | 878.6 MB | [Download KB5037331](https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=d3f5268e-3e16-4477-9fc3-538bb4353b72)        |

Feel free to reach out if you need additional information!
